lib/user.rb

```diff
@@ -0,0 +1,22 @@
+class User
+
+  attr_reader :name, :jokes
+  def initialize(name)
+    @name = name
+    @jokes = []
+  end
+
+  def learn(joke)
+    @jokes << joke
+  end
+
+  def tell(tellee, joke)
+    tellee.jokes << joke
+  end
+
+  def joke_by_id(id)
+    @jokes.each do |joke|
+      return joke if joke.id == id 
```

Comment:
this works just fine using `each`!

this is also what the `find` enumerable is exactly meant to do... 
```
@jokes.find do |joke|
      joke.id == id 
end
```